starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java

```diff
@@ -1,52 +1,35 @@
-package com.starter.fullstack.dao;
-
-import com.starter.fullstack.api.Inventory;
-import com.starter.fullstack.config.EmbedMongoClientOverrideConfig;
-import java.util.List;
-import javax.annotation.Resource;
-import org.junit.After;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.springframework.boot.test.autoconfigure.data.mongo.DataMongoTest;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito; // Library used to mock classes and methods
+import org.mockito.MockitoAnnotations;
 import org.springframework.data.mongodb.core.MongoTemplate;
-import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.junit4.SpringRunner;
 
-/**
- * Test Inventory DAO.
- */
-@ContextConfiguration(classes = {EmbedMongoClientOverrideConfig.class})
-@DataMongoTest
-@RunWith(SpringRunner.class)
 public class InventoryDAOTest {
-  @Resource
-  private MongoTemplate mongoTemplate;
-  private InventoryDAO inventoryDAO;
-  private static final String NAME = "Amber";
-  private static final String PRODUCT_TYPE = "hops";
 
-  @Before
-  public void setup() {
-    this.inventoryDAO = new InventoryDAO(this.mongoTemplate);
-  }
+    @Mock private MongoTemplate mockMongoTemplate;
```

Comment:
You shouldn't use a mock for mongo template. Utilize the mongo template set up in this test. 
Also you shouldn't remove anything from the test. Any lines here that are red should be added back. This file should only be adding to it, not removing